### ISSUE TITLE
[py2py3] Persistency pickle dump requires a string object (not bytes)

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#10569]
**Impact of the bug**
WMCore in general

**Describe the bug**
With the fix made with a new version of rucio-clients, it now exposed another string/bytes problem in the Persistency module:
```
======================================================================
ERROR: testOnWorkerNodes (WMCore_t.WMSpec_t.Steps_t.Executors_t.LogArch_t.otherLogArchiveTexst)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py", line 288, in setUp
    taskMaker.processWorkload()
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMSpec/Makers/TaskMaker.py", line 127, in processWorkload
    sandboxCreator.makeSandbox(self.workdir, self.workload)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMRuntime/SandboxCreator.py", line 138, in makeSandbox
    workload.save(workloadFile)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMSpec/Persistency.py", line 49, in save
    pickle.dump(self.data, handle)
TypeError: write() argument must be str, not bytes

----------------------------------------------------------------------
```

**How to reproduce it**
```
nosetests test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py
```

**Expected behavior**
Fix this functionality for our python3 services, while keeping it compatible with the python2 stack.

**Additional context and error message**
Can be reproduced starting in `1.4.9.pre6`

**Comment by amaltaro** [comment]
@mapellidario Dario, I do not want to push yet another issue into your plate, but given that you have completely mastered these unicode/string/bytes thing, you're probably the best candidate to work on this issue. I have the feeling this will fix many unit tests as well.

**Comment by mapellidario** [comment]
I had already put my eyes on this issue, :smirk:

**Comment by mapellidario** [comment]
As written in #10570, I would rather postpone this. I would follow #10422 and make sure that this isue is fixed by the PR that will close #10544 . If I try to fix this, I encounter a lot of problems that appear from slice1 to slice 13. I would rather fix them in a gradual way following the schedule that we have in place. I hope this is ok with all of you, especially @amaltaro !

**Comment by amaltaro** [comment]
Dario, if I understood you right, you are suggesting to fix those ad-hoc classes  (Configuration, Credential and DataStructs), then fix slices 1 all the way to 12. And only then work on this issue, which should be part of the slice 13. Is that correct?

**Comment by mapellidario** [comment]
Yes! That would be my plan! thanks!

**Comment by mapellidario** [comment]
This has been fixed by #10585, thanks Alan for the fix and for checking the status of my PR!
